### PR TITLE
Update Unit Tests for M2/M3

### PR DESCRIPTION
**Pull request by QuncCccccc**
Updated golden tests in 

* ink_sparkle_test.dart
* material_test.dart
* page_test.dart
* progress_indicator_test.dart

to have M2 and M3 versions.

More info in #127064

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by HansMuller — APPROVED** [review]
LGTM

Looks like you need to approve some golden images.

File: packages/flutter/test/material/progress_indicator_test.dart [hunk 1]
```diff
@@ -1019,7 +1035,29 @@ void main() {
...
+            padding: EdgeInsets.all(4),
+            child: CircularProgressIndicator(),
+          ),
+        ),
+      ),
+    ), const Duration(seconds: 2));
+
+    await expectLater(
+      await animationSheet.collate(20),
+      matchesGoldenFile('material.m3_circular_progress_indicator.indeterminate.png'),
```
Comment (TahaTesser) [inline packages/flutter/test/material/progress_indicator_test.dart:1060]: Above 
` matchesGoldenFile('m3_material.border_paint_below.png'),` 

but here it's `material.m3`

File: packages/flutter/test/material/progress_indicator_test.dart [hunk 2]
```diff
@@ -732,7 +732,23 @@ void main() {
...
+    await tester.pumpFrames(animationSheet.record(
+      Theme(
+          data: ThemeData(useMaterial3: true),
+          child: const _RefreshProgressIndicatorGolden()
+      ),
+    ), const Duration(seconds: 3));
+
+    await expectLater(
+      await animationSheet.collate(20),
+      matchesGoldenFile('material.m3_refresh_progress_indicator.png'),
```
Comment (TahaTesser) [inline packages/flutter/test/material/progress_indicator_test.dart:751]: Same here

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/progress_indicator_test.dart (hunk 1 above)
Comment (QuncCccccc) [reply packages/flutter/test/material/progress_indicator_test.dart]: Just updated! Thanks for helping review!

**Comment by QuncCccccc** [comment]
> LGTM
> 
> Looks like you need to approve some golden images.

Will triage them! Thanks:)

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Comment by loic-sharma** [comment]
@QuncCccccc This PR had to be reverted as it was affected by this Flutter Gold issue: https://g-issues.skia.org/issues/40045533.

See @Piinks's [explanation](https://discord.com/channels/608014603317936148/613398423093116959/1133885989358678076):

> The mirror is probably out of sync. The failing commit introduced new changes, but they have not been updated yet. I would revert the PR until we can check the mirror. Last week it went 2+ hours out of sync so rerunning it may not green the tree for a while.

**Pull request merged**